fishers-exact-test/d/dfet.d

```diff
@@ -47,51 +55,40 @@ double logHypergeometricProbability(const long[] data, const ref double[] fs) {
 		fs[data[1]] -
 		fs[data[2]] -
 		fs[data[3]] -
-		fs[data[0] + data[1] + data[2] + data[3]]
-	);
+		fs[data[0] + data[1] + data[2] + data[3]];
```

Comment:
This trick to a really fast implementation likely would come from finding an algorithm that instead of walking `i` forwards and indexing `fs` at pseudo-random (bad for cache, maybe just more work overall too), find a way to walk the `fs` forwards once. Essentially this algorithm whole thing is just a sum over `a[n] * fs[n]`, you just need to work out what `a[n]` is.

Reply:
this was nonsense. Don't know what I was thinking...